src/rust/lib/graph-editor/src/component/visualization/factory.rs

```diff
@@ -0,0 +1,149 @@
+//! Contains functionality related to creating `Visualisations` from different sources..
+//!
+//! The `Factory` provides a way to implement structs that allow the instantiation of specific
+//! visualisations, while already providing general information that doesn't require an
+//! instantiated visualization, for example, the name or input type of the visualisation.
+//!
+//! There are two example implementations: The `JsSourceFactory`, which is based on a JS snippet to
+//! instantiate `JsRenderer`, and the fairly generic `NativeConstructorFactory`, that only requires
+//! a function that can create a InstantiationResult. The later can be used as a thin wrapper around
+//! the constructor methods of native visualizations.
+//!
+//! Example
+//! --------
+//! ```
+//! use graph_editor::component::visualization::JsSourceFactory;
+//! use graph_editor::component::visualization::NativeConstructorFactory;
+//! use graph_editor::component::visualization::Metadata;
+//! use graph_editor::component::visualization::Visualization;
+//! use graph_editor::component::visualization::renderer::example::native::BubbleChart;
+//! use ensogl::display::Scene;
+//! use std::rc::Rc;
+//!
+//! // Create a factory from a JS source code snippet.
+//! let js_source_factory = JsSourceFactory::from_js_source_raw(r#"
+//! class BubbleVisualisation {
+//!     onDataReceived(root, data) {}
+//!     setSize(root, size) {}
+//! }
+//! return new BubbleVisualisation();
```

Comment:
I agree `Class` is nicer! I also changed the code to return the class object and use that instead of requiring a closure that returns an instance. This solves a few other issues, too.